logstage/logstage-sink-file/src/main/scala/com/github/pshirshov/izumi/models/SinkState.scala

```diff
@@ -0,0 +1,27 @@
+package com.github.pshirshov.izumi.models
+
+import com.github.pshirshov.izumi.models.LogFile.RealFile
+
+import scala.util.{Success, Try}
+
+
+case class SinkState(currentFileId: FileId = FileId.init
+                     , currentFileSize: Int = FileSize.undefined
+                     , files: Files = scala.collection.mutable.HashMap.empty[FileId, RealFile]
+                     , config: FileSinkConfig) {
+
+
+  def getFile(fileId: FileId): Try[RealFile] = {
+    Try(files.getOrElseUpdate(fileId, RealFile(name = fileId, path = config.destination, status = LogFileStatus.FirstEntry)))
```

Comment:
SinkState can't be used with DummyFile like that. Consider providing external means of constructing a LogFile as in:

```scala
 trait CreateLogFile[LOGFILE <: LogFile] {
    def create(path: String, name: FileId, status: LogFileStatus = LogFileStatus.FirstEntry): LOGFILE
  }

implicit val createDummyLogFile: CreateLogFile[DummyFile] = DummyFile.apply _
implicit val createRealLogFile: CreateLogFile[RealFile] = RealFile.apply _

case class FileSinkConfig(
                           logFileFactory: CreateLogFile[_ <: LogFile]
                           , fileSizeLimit: Int
                           , destination: String
                           , rotation: Rotation
                         )
```

Now SinkState can create both RealFiles and DummyFiles:

```scala
case class SinkState(
                 config: FileSinkConfig
                 , files: mutable.Map[FileId, LogFile] = mutable.HashMap.empty[FileId, LogFile]
               ) {
  def getFile(fileId: FileId): LogFile = {
    files.getOrElseUpdate(fileId, config.logFileFactory.create(config.destination, fileId, LogFileStatus.FirstEntry))
  }
}
```